test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportDataTests.cs

```diff
@@ -0,0 +1,225 @@
+        public async Task GivenFhirServer_WhenDataIsExportedWithSoftDeletesNotParallel_ThenExportedDataIsSameAsDataInFhirServer()
+        {
+            await ExportAndSoftDeleteTestHelper(parallel: false, history: false, deletes: true);
+        }
+
+        [Fact]
+        public async Task GivenFhirServer_WhenDataIsExportedWithHistoryAndSoftDeletesParallel_ThenExportedDataIsSameAsDataInFhirServer()
+        {
+            await ExportAndSoftDeleteTestHelper(parallel: true, history: true, deletes: true);
+        }
+
+        [Fact]
+        [Trait(Traits.Category, Categories.ExportLongRunning)]
+        public async Task GivenFhirServer_WhenDataIsExportedWithHistoryAndSoftDeletesNotParallel_ThenExportedDataIsSameAsDataInFhirServer()
+        {
+            await ExportAndSoftDeleteTestHelper(parallel: false, history: true, deletes: true);
+        }
+
+        // _tag filter cannot be used with history or deleted export. Using isParallel to test both SQL code paths.
+        private async Task ExportAndSoftDeleteTestHelper(bool parallel, bool history, bool deletes)
+        {
+            if (_fixture.DataStore == DataStore.CosmosDb && parallel)
+            {
+                // CosmosDB does not have parallel export support yet.
+                return;
+            }
+
+            string uniqueFixtureResources = string.Join(',', _fixture.TestResourcesWithHistoryAndDeletes.Keys.Select(x => x.resourceType).Distinct());
+            string includeAssociatedDataParam = (history ? "_history" : string.Empty) + (deletes ? (history ? "," : string.Empty) + "_deleted" : string.Empty);
+
+            // Trigger export request and check for export status. _typeFilter and history/soft delete parameters cannot be used together.
+            string parallelQueryParam = $"_isParallel={parallel}";
+            Uri contentLocation = await _fixture.TestFhirClient.ExportAsync(parameters: $"_since={_fixture.TestDataInsertionTime:O}&_type={uniqueFixtureResources}&includeAssociatedData={includeAssociatedDataParam}&{parallelQueryParam}");
+
+            IList<Uri> blobUris = await ExportTestHelper.CheckExportStatus(_testFhirClient, contentLocation);
+
+            // Download exported data from storage account
+            Dictionary<(string resourceType, string resourceId, string versionId), Resource> dataFromExport =
+                await ExportTestHelper.DownloadBlobAndParse(blobUris, _fhirJsonParser, _outputHelper);
+
+            var expectedResources = _fixture.TestResourcesWithHistoryAndDeletes;
+
+            if (!history)
+            {
+                expectedResources = _fixture.TestResourcesWithDeletes;
+            }
+
+            if (!deletes)
+            {
+                expectedResources = _fixture.TestResourcesWithHistory;
+            }
+
+            // Assert both data are equal
+            Assert.True(ExportTestHelper.ValidateDataFromBothSources(_fixture.TestResourcesWithHistoryAndDeletes, dataFromExport, _outputHelper));
```

Comment:
You're setting the expected resources and then not using that variable.

Reply:
Thanks!! The tests didn't fail in the PR because they are "long running" tests. Good eye.